### DISCUSSION TITLE
[C/CPP] parsing bug

### DISCUSSION THREAD

**Discussion by Tomotz** [github/securitylab#269] [Q&A] [answered]
Hi, I've written a small test code to check codeQL:
![code](https://user-images.githubusercontent.com/12498248/107141847-2c74cc80-6934-11eb-8555-a100bc027591.png)

I've run some queries on the code, and was missing some results, so I tried checking the AST:
![AST](https://user-images.githubusercontent.com/12498248/107141868-34347100-6934-11eb-8330-dc75deb324f9.png)

I saw that any node that involves the variable 'i' was errored in the AST (ErrorExp).
I assume the errors are caused from this variable declaration which happens inside the for loop.
Is this a bug, or did I do something wrong?

I am using the 'vscode-codeql-starter' workspace downloaded from codeQL website.

Thanks

**Comment by criemen** [comment, marked as answer]
Hi Tomotz,

this looks like a bug in our code!
In order for us to debug this, I have a couple of questions:
- Can you share the source code for your test?
- How did you build the database from your source code (please include the exact `codeql database create` invocation)
- What is the version of the CodeQL CLI you're using? Just copy-paste the output of `codeql --version`. 

Thank you very much.


Edit: See discussion below, this is caused by a known bug.

**Reply by Tomotz** [reply]
Source code:
[Archive.zip](https://github.com/github/securitylab/files/5943711/Archive.zip)

codeql database create v2 --language=cpp --command="make"


"""
CodeQL command-line toolchain release 2.4.3.
Copyright (C) 2019-2021 GitHub, Inc.
Unpacked in: /Users/tom/Desktop/work/MyCode/CodeQL/codeql
   Analysis results depend critically on separately distributed query and
   extractor modules. To list modules that are visible to the toolchain,
   use 'codeql resolve qlpacks' and 'codeql resolve languages'.
"""

Thanks!

**Reply by criemen** [reply]
Hi, checked on both Linux and osx and cannot reproduce the error.
Can you zip up the database that has the error and upload it?

Thanks a lot in advance.

**Reply by Tomotz** [reply]
[v3.zip](https://github.com/github/securitylab/files/5951504/v3.zip)

I'm running in macOS btw

**Reply by criemen** [reply]
You're unfortunately encountering a known bug.
You can work around it by adding `-std=c99` to the clang command line.

Then we parse your source code correctly.
